pkg/kvstore/etcd.go

```diff
@@ -694,25 +697,33 @@ func connectEtcdClient(ctx context.Context, config *client.Config, cfgPath strin
 		statusCheckErrors:    make(chan error, 128),
 	}
 
+	handleSessionError := func(err error) {
+		ec.RWMutex.Lock()
+		ec.sessionErr = err
+		ec.RWMutex.Unlock()
+		errChan <- err
+	}
+
 	// wait for session to be created also in parallel
 	go func() {
 		defer close(errChan)
+		defer close(ec.firstSession)
```

Comment:
Why the extra error close here?

Reply:
good point, this will be `errChan` so this will likely panic as we are closing a closed channel.

Reply:
The commit disconnects `errChan` and `firstSession` again as we can't return an error. It has to be closed. Prior to  bf8e432 it was only closed on successful session establishment which meant it was leaked and with it all code that blocked on it, with this commit, it is always closed at the end of `connectEtcdClient()` and any user of `firstSession` relying on a valid etcd client has to check `sessionError()`.

Reply:
ah, right, it was a oversight on my side. Is that what you were thinking @joestringer ?

Reply:
:+1: yep and on first read I missed that we actually create a new channel here rather than reusing `errChan`. LGTM.